### ISSUE TITLE
I get an ERROR TypeError: Cannot read property 'forEach' of null...

### ISSUE THREAD

**Issue by cunundrummer** [ngx-formly/ngx-formly#1530]
**I'm submitting a ...**  (check one with "x")
```
[ x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ x] support request
```

**Current behavior**

Not too sure how the bug (2 errors - keep reading) manifests.  But it has something to do when assigning the fields in the formly-form... 

```
<form [formGroup]="universalForm" (ngSubmit)="submit()">
  <formly-form [model]="model" [fields]="allFields$ | async" [options]="options" [form]="universalForm"></formly-form>
  <button type="submit" class="btn btn-primary submit-button" [disabled]="!universalForm.valid">Submit</button>
  <button type="button" class="btn btn-default" (click)="options.resetModel()">Reset</button>
</form>
```
...asynchronously in the html file; something to do with loading the form.  All is good until I do a forkjoin operation. ex.

```
console.log('no error here');
this.allFields$ = forkJoin(obs1, obs2)
      .pipe(
        map(([first, second]) => {
          console.log({first});
          console.log({second});
          const fieldList = first.concat(second);
          console.log(fieldList); // is now a proper formlyFieldConfig[]
          return fieldList;
        })
      );
```
I have console.logged and there is no error until the allFields$.forkJoin assignment. After I console.log('no error here') I get the error in the console:

> ERROR TypeError: Cannot read property 'forEach' of null
    at FormlyFormExpression.push../node_modules/@ngx-formly/core/fesm5/ngx-formly-core.js.FormlyFormExpression._checkFields (ngx-formly-core.js:625)

I did try to put it in afterViewInit, but still same issue. Also, I saw that only undefined is being inspected, but not null:

```
FormlyFormExpression.prototype._checkFields = /**
     * @param {?} form
     * @param {?=} fields
     * @param {?=} model
     * @param {?=} options
     * @return {?}
     */
    function (form, fields, model, options) {
        var _this = this;
        if (fields === void 0) { fields = []; } /****line 625****/
        fields.forEach(function (field) {
            _this.checkFieldExpressionChange(form, field, _this.getParentModel(model, field), options);
            _this.checkFieldVisibilityChange(form, field, _this.getParentModel(model, field), options);
            if (field.fieldGroup && field.fieldGroup.length > 0) {
                _this._checkFields(field.formControl ? /** @type {?} */ (field.formControl) : form, field.fieldGroup, _this.getParentModel(model, field), options);
            }
        });
    };
```

Maybe do a if (fields == null) or a combination of (fields === void 0 || fields === null) might help?  
I fiddled a bit more and it _seems_ to be an asynchronous issue.  If I have

`fields = [];` 

and I have [fields]=fields, no errors nor any form fields(as expected).  

In the end, I do get the results that I want oddly enough.  I just want to avoid any errors in the console.

<img width="1440" alt="Screen Shot 2019-04-16 at 1 07 35 AM" src="https://user-images.githubusercontent.com/5727752/56183336-246fce00-5fe4-11e9-90b4-f7a5d4972189.png">

**Expected behavior**

Everything the same but without the errors. (see above image for the errors)

**What is the motivation / use case for changing the behavior?**

I get fields configurations from more than 1 source(s)

**Please tell us about your environment:**

* **Angular version:** 7.2.0
* **Formly version: ** "@ngx-formly/core": "^4.8.2",

**Comment by cunundrummer** [comment]
Thanks again for the quick response.  But I hate to say it, it is now definitely a asynchronous issue.  I reinstalled formly.  

When I do [fields]="allFields$ | async" in the formly-form element, I still get an error.  Depending on when I get the values from my datasources (I think), I get errors that have to do with for-each of undefined.

At the moment, my workaround, is to subscribe directly in the components .ts file as opposed to the components html file.  ex.

`  ...<formly-form [model]="model" [fields]="allFields" [options]="options" [form]="universalForm"></formly-form>...
`
and

```
    this.allFields$ = forkJoin(obs1, obs2)
      .pipe(
        map(([first, second]) => {
          console.log({first});
          console.log({second});
          const fieldList = first.concat(second);
          console.log(fieldList);
          return fieldList;
        })
      );
    this.allFieldsSubs$ = this.allFields$.subscribe((fields: FormlyFieldConfig[]) => {
      this.allFields = fields; // fix to remove errors from formly (has trouble with  (observable|async) ops)
    });
```

I have it working without errors for now.  This is just an informational heads up and for anyone else whom has similar issues.  Keep up the good work!

**Comment by aitboudad** [comment]
fixed in `5.0.4`, you can pass `null` or `undefined` to `fields`.
